### ISSUE TITLE
Bug in lineplot

### ISSUE THREAD

**Issue by Castau** [Hold-Krykke/PythonExam#18]
Something is wrong in the lineplot regarding the dates. 
Malte has seen this error before. Something about the expected date being converted to a datetime. 

![image](https://user-images.githubusercontent.com/35559774/81714094-fd016400-9476-11ea-9f5f-d32309fa8bed.png)

**Comment by MalteMagnussen** [comment]
The image link is dead for me.

**Comment by MalteMagnussen** [comment]
What are the steps to reproduce this error? @HrBjarup

**Comment by MrHardcode** [comment]
The bug occurs on this branch: #13 Check out the latest commit on the branch and run `flask_service.py`. Open Postman and do a POST request to the following endpoint `http://localhost:5000/api/sentiment` with the following body: 
```JSON
{
    "hashtags": [
        "trump",
        "biden"
    ],
    "start_date": "2020-5-8",
    "end_date": "2020-5-12",
    "plot_type": "line"
}
```
This should result in the following plot:
![image](https://user-images.githubusercontent.com/44898491/81739178-2a600900-949b-11ea-9506-e6d8da5f5cc5.png)

The bug doesn't occur when using a `bar` plot instead and I have no idea why
![image](https://user-images.githubusercontent.com/44898491/81739415-9c385280-949b-11ea-8998-9c79041b14a9.png)

**Comment by MalteMagnussen** [comment]
Trying it now. 

Prøvede at putte en guard ind for at tjekke om det faktisk var datetime.date der kom med
![image](https://user-images.githubusercontent.com/11192971/81746725-42d62080-94a7-11ea-826c-9f0e061683de.png)
![image](https://user-images.githubusercontent.com/11192971/81746759-4a95c500-94a7-11ea-90fb-0396f745af2a.png)
Men buggen var der stadig. 

datetime.date har nemlig kun year, month, day, og ikke tid. så hvordan den viser tid er virkelig underligt 


### **POSSIBLE ROOT OF ISSUE FOUND**

If you select too few dates, it squeezes in some time, to make sure there are at least 6 xlabels 
![image](https://user-images.githubusercontent.com/11192971/81747107-d4de2900-94a7-11ea-8c3e-7c4b3419c347.png)

**At 5 days its bad:**
![image](https://user-images.githubusercontent.com/11192971/81747183-ecb5ad00-94a7-11ea-9e03-e7675f90f93a.png)
**At 6 its fine:** 
![image](https://user-images.githubusercontent.com/11192971/81747203-f63f1500-94a7-11ea-80c5-f6ab7472b065.png)
![image](https://user-images.githubusercontent.com/11192971/81747396-546bf800-94a8-11ea-90b6-b5dbdd47cd84.png)


**POSSIBLE HACKY SOLUTION**

Only allow user to select 6 days or more when doing line graphs lol

(find a real solution?)

**Comment by MalteMagnussen** [comment]
### **Det her er faktisk et andet issue**

I forbindelse med at den orange bare stopper. Ønsket behavior er at den går til 0. Men det er pga NaN i dict. 
![image](https://user-images.githubusercontent.com/11192971/81745966-fdfdba00-94a5-11ea-8828-e9575fefa098.png)

**Løsning:**
fint, så kan man skrive df.fillna(value=0, inplace=True) så er den klaret
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.fillna.html
@Runi-VN Fandt denne løsning

Gør det på denne DataFrame 
![image](https://user-images.githubusercontent.com/11192971/81745874-df97be80-94a5-11ea-9849-087db40edd2e.png)

**Comment by MrHardcode** [comment]
Plotting is still bugged - check out [this](https://github.com/Hold-Krykke/PythonExam/blob/174aae59855b7acd1c6d94a0d0652e1806cd6bbe/modules/plots/%40JoeBiden.png)

**Comment by MalteMagnussen** [comment]
> Plotting is still bugged - check out [this](https://github.com/Hold-Krykke/PythonExam/blob/174aae59855b7acd1c6d94a0d0652e1806cd6bbe/modules/plots/%40JoeBiden.png)

This was related to the naming of the @ thing on the dict, right? I had named it differently than Runi

**Comment by MalteMagnussen** [comment]
Fixed in #21
